### ISSUE TITLE
Get all home charging sessions

### ISSUE THREAD

**Issue by HaraldWalter** [tillsteinbach/WeConnect-python#32]
I would like to export all home charging sessions to evaluate how much energy was taken from which wallbox and authenticated by which charging card. 

So far I am working with the iobroker.vw-connect and iobroker.simple-api adapter. I am requesting the data from a python script using the simple-api REST API. This is not very efficient but was working in a first POC. I would like to get rid of the iobroker dependency providing an exporting possibility that is kind of standalone. It would be great to use the weconnect-python module to handle the VW API. However, I think there is no API yet to return the former home charging sessions . Would it be possible to add it?

**Comment by tillsteinbach** [comment]
You mean the data from the Elli Chargers?

**Comment by HaraldWalter** [comment]
yes

**Comment by tillsteinbach** [comment]
Ok, that is a good request, I already had that in mind to allow automatically to connect charging sessions in [VWsFriend](https://github.com/tillsteinbach/VWsFriend) with the data from WeCharge. That would be the same authentication. The problem for home charging is that I do not have such a charger. So once the groundwork is done I would need some data examples. But First of all I would need some refactoring so.
To summarise: I can imagine to implement that but it is a bigger part and will need some time.

**Comment by waltster3** [comment]
Hi, 

we have three ID Chargers with 4 different RFID charging cards. We already generated 300+ records and I would share the data. Is there something else I could contribute (question also relates to source code)?

**Comment by tillsteinbach** [comment]
I guess it is also json what is coming from the server.
The WeCharge items need a different authentication, thus I will first refactor out the current WeConnect auth, (login, token refresh, ...) in a separate class. Then I can add the auth for the charging. The actual retrieval of the data should be a piece of cake, but it would be helpful to have a sample of the data in json format if you can capture that.

**Comment by tillsteinbach** [comment]
Short update: I was able to login to WeCharge the first time and fetch some data. I still have a lot of cleanup todo before I can think of integrating it into a release, but a first proof of concept is done.

**Comment by waltster3** [comment]
I just have the json string coming from io-broker. However, I am not sure if this is the same format than returned by WeCharge.

**Comment by tillsteinbach** [comment]
We can have a try. You can anonymize it if it contains personal data.

**Comment by waltster3** [comment]
Sure!

100 records with anonymized serial numbers, IDs and addresses. 

[wecharge_anonym.zip](https://github.com/tillsteinbach/WeConnect-python/files/7963289/wecharge_anonym.zip)

**Comment by tillsteinbach** [comment]
super danke! Wird noch ein paar tage dauern, ich muss erst die neue Authentifizierung einbauen, aber ich bin dran versprochen!

**Comment by Peter1984Git** [comment]
Any update on this? Elli Charger now supports EEBUS and VW Passat GTE is here (still waiting for ID.4...)

**Comment by tillsteinbach** [comment]
I have a working example, unfortunately I did not find enough time to finish it yet. I would need one or two days. Unfortunately my real job and family keeps me busy right now. But ist is not forgotten.

**Comment by waltster3** [comment]
> Any update on this? Elli Charger now supports EEBUS and VW Passat GTE is here (still waiting for ID.4...)

Is the EEBUS update finally available? I am waiting for the respective Fota update since 1,5 years meanwhile…

**Comment by Peter1984Git** [comment]
> > Any update on this? Elli Charger now supports EEBUS and VW Passat GTE is here (still waiting for ID.4...)
> 
> Is the EEBUS update finally available? I am waiting for the respective Fota update since 1,5 years meanwhile…

Yes, it is. You can install it via VW Connect ID App or wait for Autoupdate.

**Comment by Schnabel80** [comment]
> I have a working example, unfortunately I did not find enough time to finish it yet. I would need one or two days. Unfortunately my real job and family keeps me busy right now. But ist is not forgotten.

Hi @tillsteinbach have you any news about the ID Chargers?

**Comment by tillsteinbach** [comment]
Unfortunately not. Got even worse with spare time. Sorry.

**Comment by Schnabel80** [comment]
> Unfortunately not. Got even worse with spare time. Sorry.

we all have the same problems...
thank you for your great work!

**Comment by tillsteinbach** [comment]
Yes, I was able to get the charging sessions.

**Comment by MichaelHuelsen** [comment]
Hey there @tillsteinbach  / @ksokol  / @waltster3 

I am having a similar use case, my wallbox is connected to the Volkswagen app and my charging sessions appear in the WeCharge integration of the app.  I was also able to retrieve the charging history including date filtering as well as individual charing sessions, I used the CarConnectivity library / modules for the login procedure.

Since WeConnect-python will be discontinued, I am wondering if we could integrate this into the CarConnectivity project. @tillsteinbach  please advice how we may proceed, should it belong somewhere in the CarConnectivity-connector-volkswagen repository?
